### DISCUSSION TITLE
How to persist user session in a client app?

### DISCUSSION THREAD

**Discussion by emailvenkatm** [appwrite/appwrite#3938] [Q&A] [answered]
### 💭 Description

We are building a mobile app in Flutter for which we are using apprwrite as Backend system. So here is what we want to achieve. When the user is logged in and killed the app, the app should not ask the user to re-login once it is reopened. Please advice.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by bliyscom** [comment]
If I am allowed to respond, I would like to give you guys a suggestion. Currently, to the best of my knowledge, appwrite only supports a token refresh if you used oAuth to create the session so this is what i suggest:
1. Use and embedded database like Hive to store the users login credentials in their device. Hive supports encryption so use it to make it a secure storage.
2.  Create a session create method and run it in the initstate of your landing page so that everytime the app is killed and re-opened, the user should be automatically logged-in.
3.  To avoid creating multiple sessions for the same user, run a delete all session call just before the new session create call inside the session create method mentioned above.
4.  Lastly, I would want to mention the security risk of such a functionality as it means someone can pick up your user;s phone and have access to the app without permission since it is auto-login. Therefore, it should be an op-tin funcionality. 
5. You can make the functionality op-tin by using a provider to save the state of "whether to auto-login or not" and save the provider value in hive or sharedpreferences to persist it across logins. You can also implement biometric logins to beef up security.

**Comment by stnguyen90** [comment, marked as answer]
@emailvenkatm, actually, if you're using our Flutter SDK, the session lasts for 1 year and is automatically persisted to application data. Regardless of which SDK you use, your app should do the following on open:

1. initialize the client and services
2. call account.get()
3. if account.get() throws an exception, direct user to login page
4. if account.get() is successful, save the result in global state and proceed to the home page of your app

Make sure there is only one instance of client and it is reused across your app. If you have any problems, feel free to reach out on [Discord](https://appwrite.io/discord).

**Reply by notsarv** [reply]
This will also throw an exception if the device is not connected to internet ??
